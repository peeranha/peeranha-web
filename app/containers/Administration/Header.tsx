import { useTranslation } from 'react-i18next';
import React from 'react';

import { MediumIconStyled } from 'components/Icon/MediumIcon';
import Wrapper from 'components/Header/Simple';
import Icon from 'components/Icon';
import H3 from 'components/H3';

import AddRoleForm from 'containers/Administration/AddRoleForm';
import { Moderator } from 'containers/Administration/types';

import AddModeratorButton from 'containers/Administration/AddModeratorButton';

import { singleCommunityColors } from 'utils/communityManagement';
import { BORDER_PRIMARY, ICON_TRASPARENT_BLUE } from 'style-constants';
import { Administration } from 'icons/index';
import { css } from '@emotion/react';

const colors = singleCommunityColors();

const customColor = colors.linkColor || BORDER_PRIMARY;

type HeaderProps = {
  locale: string;
  single?: number;
  moderators: Array<Moderator>;
  addRole: (userAddress: string, role: number, communityId: number) => void;
  addRoleLoading: boolean;
};

export const Header: React.FC<HeaderProps> = ({
  locale,
  single,
  moderators,
  addRole,
  addRoleLoading,
}): JSX.Element => {
  const { t } = useTranslation();

  return (
    <Wrapper className="mb-to-sm-0 mb-from-sm-3">
      <H3>
        <div
          css={css`
            .fill {
              fill: ${customColor};
            }
            .stroke {
              stroke: ${customColor};
            }

            .semitransparent {
              fill: ${colors.transparentIconColor || ICON_TRASPARENT_BLUE};
            }
          `}
        >
          <MediumIconStyled>
            <Administration />
          </MediumIconStyled>
        </div>
        {t('common.administration')}
      </H3>
      <AddRoleForm
        locale={locale}
        single={single}
        addRole={addRole}
        moderators={moderators}
        Button={AddModeratorButton}
        addRoleLoading={addRoleLoading}
      />
    </Wrapper>
  );
};

export default React.memo(Header);
