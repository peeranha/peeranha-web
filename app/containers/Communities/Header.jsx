import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import * as routes from 'routes-config';

import commonMessages from 'common-messages';

import createCommunityIcon from 'images/createCommunity.svg?inline';
import addIcon from 'images/add.svg?external';

import Icon from 'components/Icon';
import TransparentButton from 'components/Button/Contained/Transparent';
import SubHeaderWrapper from 'components/Header/Complex';
import NavigationButton from 'components/Button/Contained/Navigation';
import A from 'components/A';

import messages from './messages';
import languages from './languagesOptions';
import { GO_TO_CREATE_COMMUNITY_SCREEN_BUTTON_ID } from './constants';

const suggestedCommunitiesRoute = routes.suggestedCommunities();
const communitiesRoute = routes.communities();

const Header = ({
  goToCreateCommunityScreen,
  SubHeader,
  changeSorting,
  sorting,
  communitiesNumber,
  setLang,
  language,
}) => {
  const path = window.location.pathname + window.location.hash;

  return (
    <div className="mb-to-sm-0 mb-from-sm-3">
      <SubHeaderWrapper position="top">
        <div>
          <A to={communitiesRoute} href={communitiesRoute}>
            <NavigationButton isLink={path !== communitiesRoute}>
              <FormattedMessage {...commonMessages.communities} />
            </NavigationButton>
          </A>

          <A to={suggestedCommunitiesRoute} href={suggestedCommunitiesRoute}>
            <NavigationButton isLink={path !== suggestedCommunitiesRoute}>
              <FormattedMessage {...messages.voting} />
            </NavigationButton>
          </A>
        </div>

        <div className="right-panel">
          <TransparentButton
            id={`${GO_TO_CREATE_COMMUNITY_SCREEN_BUTTON_ID}_header`}
            onClick={goToCreateCommunityScreen}
            className="d-flex align-items-center"
          >
            <img
              className="d-none d-sm-inline-block"
              src={createCommunityIcon}
              alt="icon"
            />

            <Icon
              className="d-inline-block d-sm-none"
              icon={addIcon}
              width="14"
            />

            <span className="ml-1">
              <FormattedMessage {...messages.suggestCommunity} />
            </span>
          </TransparentButton>
        </div>
      </SubHeaderWrapper>

      <SubHeader
        changeSorting={changeSorting}
        sorting={sorting}
        communitiesNumber={communitiesNumber}
        setLang={setLang}
        language={language}
        languages={languages}
      />
    </div>
  );
};

Header.propTypes = {
  goToCreateCommunityScreen: PropTypes.func,
  SubHeader: PropTypes.any,
  changeSorting: PropTypes.func,
  sorting: PropTypes.object,
  communitiesNumber: PropTypes.number,
  setLang: PropTypes.func,
  language: PropTypes.object,
};

export default Header;
