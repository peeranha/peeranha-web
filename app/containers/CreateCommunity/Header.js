import React from 'react';
import { FormattedMessage } from 'react-intl';

import * as routes from 'routes-config';
import commonMessages from 'common-messages';
import { TEXT_PRIMARY } from 'style-constants';

import createCommunityHeader from 'images/communitiesHeader.svg?inline';
import closeIcon from 'images/closeCircle.svg?inline';

import A from 'components/A';
import H3 from 'components/H3';
import Span from 'components/Span';
import BaseRounded from 'components/Base/BaseRounded';
import { MediumImageStyled } from 'components/Img/MediumImage';

import messages from './messages';

export const Header = /* istanbul ignore next */ () => (
  <BaseRounded className="d-flex align-items-center justify-content-between mb-3">
    <H3 className="d-flex align-items-end">
      <MediumImageStyled src={createCommunityHeader} alt="questionIcon" />
      <FormattedMessage {...messages.newCommunity} />
    </H3>

    <A
      to={routes.communities()}
      href={routes.communities()}
      className="d-inline-flex align-items-center"
    >
      <img className="mr-1" src={closeIcon} alt="x" />
      <Span color={TEXT_PRIMARY}>
        <FormattedMessage {...commonMessages.close} />
      </Span>
    </A>
  </BaseRounded>
);

export default React.memo(Header);
