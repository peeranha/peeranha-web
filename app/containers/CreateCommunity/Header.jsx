import React from 'react';
import { FormattedMessage } from 'react-intl';

import * as routes from 'routes-config';
import commonMessages from 'common-messages';
import { TEXT_PRIMARY } from 'style-constants';

import createCommunityHeader from 'images/communitiesHeader.svg?inline';
import closeIcon from 'images/closeCircle.svg?inline';

import A from 'components/A';
import H3 from 'components/H3';
import Span from 'components/Span';
import Wrapper from 'components/Header/Simple';
import { MediumImageStyled } from 'components/Img/MediumImage';

import messages from './messages';

export const Header = () => (
  <Wrapper className="mb-to-sm-0 mb-from-sm-3">
    <H3>
      <MediumImageStyled src={createCommunityHeader} alt="create-community" />
      <FormattedMessage {...messages.newCommunity} />
    </H3>

    <div className="right-panel">
      <A to={routes.communities()}>
        <button>
          <img className="mr-1" src={closeIcon} alt="x" />
          <Span color={TEXT_PRIMARY}>
            <FormattedMessage {...commonMessages.close} />
          </Span>
        </button>
      </A>
    </div>
  </Wrapper>
);

export default React.memo(Header);
