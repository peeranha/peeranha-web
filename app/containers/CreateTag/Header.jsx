import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import * as routes from 'routes-config';
import { TEXT_PRIMARY, BORDER_PRIMARY } from 'style-constants';

import suggestTagIcon from 'images/tagsHeaderIcon.svg?external';
import closeIcon from 'images/closeCircle.svg?external';

import A from 'components/A';
import H3 from 'components/H3';
import Span from 'components/Span';
import Wrapper, { WrapperRightPanel } from 'components/Header/Simple';
import Icon from 'components/Icon';
import { IconMd } from 'components/Icon/IconWithSizes';
import { MediumIconStyled } from 'components/Icon/MediumIcon';

import { singleCommunityColors } from 'utils/communityManagement';

const colors = singleCommunityColors();

export const Header = ({
  title,
  closeRedirectPage,
  closeButtonAction = null,
}) => {
  const { t } = useTranslation();

  return (
    <Wrapper className="mb-to-sm-0 mb-from-sm-3">
      <H3>
        <MediumIconStyled>
          <Icon
            icon={suggestTagIcon}
            width="43"
            fill={colors.btnColor || BORDER_PRIMARY}
            isColorImportant
          />
        </MediumIconStyled>
        {title || t('tags.newTag')}
      </H3>

      <WrapperRightPanel className="right-panel">
        <A
          to={closeRedirectPage || routes.communities()}
          onClick={closeButtonAction}
        >
          <button>
            <IconMd
              className="mr-1"
              icon={closeIcon}
              fill={colors.btnColor || BORDER_PRIMARY}
              isColorImportant
            />
            <Span
              color={colors.btnColor || TEXT_PRIMARY}
              className="button-label"
            >
              {t('common.close')}
            </Span>
          </button>
        </A>
      </WrapperRightPanel>
    </Wrapper>
  );
};

Header.propTypes = {
  title: PropTypes.element,
  closeRedirectPage: PropTypes.string,
  closeButtonAction: PropTypes.func,
};

export default React.memo(Header);
