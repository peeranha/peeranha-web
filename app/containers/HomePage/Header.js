import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';

import {
  LANDING_FONT,
  TEXT_PRIMARY,
  TEXT_LIGHT,
  TEXT_SECONDARY,
} from 'style-constants';

import logo from 'images/Logo.svg?inline';
import login from 'images/Login.svg?external';
import closeIcon from 'images/close.svg?external';
import headerNavigation from 'images/headerNavigation.svg?external';
import bgLogin from 'images/BG_Login.png';

import { scrollToSection } from 'utils/animation';
import * as routes from 'routes-config';
import ModalDialog from 'containers/ModalDialog';

import A from 'components/A';
import Icon from 'components/Icon';
import Button from 'components/Button/Outlined/InfoStretching';
import IconStyled, { IconHover } from 'components/Icon/IconStyled';

import { showHeaderPopup, closeHeaderPopup } from './actions';
import * as homepageSelectors from './selectors';

import {
  HEADER_ID,
  FIRST_SCREEN,
  SECOND_SCREEN,
  THIRD_SCREEN,
  FOURTH_SCREEN,
  FIFTH_SCREEN,
  SEND_EMAIL_FORM_HEADER,
} from './constants';

import messages from './messages';
import EmailLandingForm from './EmailLandingForm';

/* eslint jsx-a11y/no-noninteractive-element-interactions: 0 */

export class Header extends React.PureComponent {
  state = {
    isToggled: false,
  };

  toggle /* istanbul ignore next */ = screen => {
    this.setState({ isToggled: !this.state.isToggled });
    scrollToSection(typeof screen === 'string' ? `#${screen}` : ``);
  };

  render() /* istanbul ignore next */ {
    const {
      showHeaderPopupDispatch,
      closeHeaderPopupDispatch,
      translations,
      sendEmail,
      sendEmailLoading,
    } = this.props;

    const { isToggled } = this.state;

    return (
      <Wrapper>
        <Box id={HEADER_ID} isToggled={isToggled}>
          <ModalDialog
            show={this.props.showPopup}
            closeModal={closeHeaderPopupDispatch}
          >
            <div className="header-modal-dialog">
              <div className="image-coins">
                <img src={bgLogin} alt="bgLogin" />
              </div>

              <div className="close-icon" onClick={closeHeaderPopupDispatch}>
                <Icon icon={closeIcon} />
              </div>

              <div>
                <p className="modal-dialog-message">
                  <FormattedMessage {...messages.platformUnderDeveloping} />
                </p>
                <EmailLandingForm
                  form={SEND_EMAIL_FORM_HEADER}
                  button={messages.getReward}
                  sendEmail={sendEmail}
                  sendEmailLoading={sendEmailLoading}
                  translations={translations}
                />
              </div>
            </div>
          </ModalDialog>

          <div className="container">
            <div className="row">
              <div className="col-6 col-lg-4 col-xl-5 logo">
                <A to={routes.home()} href={routes.home()}>
                  <img
                    onClick={() => scrollToSection(`#${FIRST_SCREEN}`)}
                    src={logo}
                    alt="logo"
                  />
                </A>
              </div>

              <button
                className="col-6 d-inline-block d-lg-none navbar-toggler"
                type="button"
                onClick={this.toggle}
              >
                <Icon icon={isToggled ? closeIcon : headerNavigation} />
              </button>

              <div
                className={`col-md-12 col-lg-8 col-xl-7 ${!isToggled &&
                  'd-none'} d-lg-block navbar`}
              >
                <div className="row">
                  <div className="col-12 col-lg-7">
                    <div className="row navigation">
                      <span
                        className="col-md-12 col-lg-3"
                        onClick={() => this.toggle(SECOND_SCREEN)}
                      >
                        <FormattedMessage {...messages.about} />
                      </span>

                      <span
                        className="col-md-12 col-lg-3"
                        onClick={() => this.toggle(THIRD_SCREEN)}
                      >
                        <FormattedMessage {...messages.rewards} />
                      </span>

                      <span
                        className="col-md-12 col-lg-3"
                        onClick={() => this.toggle(FOURTH_SCREEN)}
                      >
                        <FormattedMessage {...messages.faq} />
                      </span>

                      <span
                        className="col-md-12 col-lg-3"
                        onClick={() => this.toggle(FIFTH_SCREEN)}
                      >
                        <FormattedMessage {...messages.team} />
                      </span>
                    </div>
                  </div>

                  <div className="col-md-12 col-lg-5">
                    <div className="row">
                      <button
                        className="col-md-12 col-lg-6 log-in-button"
                        onClick={showHeaderPopupDispatch}
                      >
                        <Icon icon={login} />
                        <FormattedMessage {...messages.login} />
                      </button>

                      <Button
                        className="col-md-12 col-lg-6"
                        onClick={showHeaderPopupDispatch}
                      >
                        <FormattedMessage {...messages.signUpFree} />
                      </Button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Box>
      </Wrapper>
    );
  }
}

/* eslint indent: 0 */
/* eslint jsx-a11y/click-events-have-key-events: 0 */
/* eslint jsx-a11y/no-static-element-interactions: 0 */

const Box = styled.div`
  * {
    outline: none !important;
    letter-spacing: -0.9px;
    font-family: ${LANDING_FONT};
    font-weight: 400;
  }

  position: relative;
  top: 0;
  width: 100%;
  background-color: rgba(9, 17, 40, 0.8);
  z-index: 9999;
  padding: 19px 0;

  .logo {
    img {
      width: 240px;
      text-align: left;
    }

    display: flex;
    padding-top: 7px;
  }

  .navbar {
    font-size: 16px;
    padding-top: 12px;
    padding-bottom: 12px;

    > .row {
      margin: 0;
    }

    .row,
    .row button,
    .row div {
      height: 100%;
      align-items: center;
    }

    .navigation span,
    .log-in-button {
      display: inline-block;
      text-align: center;
      color: ${TEXT_LIGHT};
      cursor: pointer;

      ${IconStyled} {
        margin-right: 9px;
        width: 21px;
        height: 21px;
      }

      :hover {
        color: ${TEXT_PRIMARY};

        ${IconHover} {
          ${IconHover({ color: TEXT_PRIMARY })};
        }
      }
    }

    .log-in-button {
      text-align: left;
    }
  }

  .header-modal-dialog {
    max-width: 480px;
    min-height: 280px;
    display: flex;
    justify-content: center;
    align-items: flex-end;
    padding: 50px 20px 10px 20px;

    * {
      font-family: ${LANDING_FONT};
      font-size: 18px;
    }

    .close-icon {
      position: absolute;
      right: 0px;
      top: 0px;
      padding: 15px 5px;

      ${IconStyled} {
        ${IconHover({ color: TEXT_SECONDARY })};

        width: 18px;
      }
    }

    .modal-dialog-message {
      text-align: left;
      font-size: 17px;
      padding-bottom: 30px;
    }

    .image-coins {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;

      img {
        width: 100%;
      }
    }
  }

  @media only screen and (max-width: 992px) {
    text-align: center;
    padding: 11px 0;
    position: fixed;
    background-color: #17234a !important;
    height: ${/* istanbul ignore next */ x => (x.isToggled ? `100vh` : `auto`)};

    > div {
      .logo img {
        width: 180px;
      }
    }

    .navbar-toggler {
      text-align: right;

      ${IconStyled} {
        ${IconHover({ color: TEXT_LIGHT })};

        width: 18px;
      }
    }

    .navbar {
      .navigation span,
      button {
        min-height: calc(100vh / 8);
        display: flex;
        justify-content: center;
        align-items: center;
      }

      .row,
      .row button,
      .row div {
        height: auto;
        justify-content: center;
        margin: 0;
      }
    }

    ${/* istanbul ignore next */ x =>
      x.isToggled
        ? `
      .navbar {
        display: flex !important;
        justify-content: center;

        .log-in-button {
          text-align: center;
        }
      }
    `
        : ``};
  }
`;

const Wrapper = styled.header`
  .scroll {
    position: fixed;
    background-color: rgba(9, 17, 40, 0.9);
    animation: header 1s;

    @keyframes header {
      0% {
        transform: translate(0px, -180px);
      }
      100% {
        transform: translate(0, 0px);
      }
    }

    padding: 5px 0;
    > div .logo {
      img {
        width: 180px !important;
      }
    }
  }
`;

Header.propTypes = {
  sendEmailLoading: PropTypes.bool,
  sendEmail: PropTypes.func,
  showHeaderPopupDispatch: PropTypes.func,
  closeHeaderPopupDispatch: PropTypes.func,
  showPopup: PropTypes.bool,
  popupPosition: PropTypes.object,
  translations: PropTypes.object,
  location: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  showPopup: homepageSelectors.selectShowPopup(),
  popupPosition: homepageSelectors.selectHeaderPopupPosition(),
});

export function mapDispatchToProps(dispatch) /* istanbul ignore next */ {
  return {
    dispatch,
    showHeaderPopupDispatch: () => dispatch(showHeaderPopup()),
    closeHeaderPopupDispatch: () => dispatch(closeHeaderPopup()),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Header);
