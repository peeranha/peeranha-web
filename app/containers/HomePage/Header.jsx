import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';

import { LANDING_FONT, TEXT_PRIMARY, TEXT_LIGHT } from 'style-constants';
import createdHistory from 'createdHistory';
import * as routes from 'routes-config';

import logo from 'images/Logo.svg?inline';
import login from 'images/Login.svg?external';
import closeIcon from 'images/close.svg?external';
import headerNavigation from 'images/headerNavigation.svg?external';

import { scrollToSection } from 'utils/animation';

import A from 'components/A';
import Icon from 'components/Icon';
import Button from 'components/Button/Outlined/InfoLarge';
import { IconHover } from 'components/Icon/IconStyled';

import {
  HEADER_ID,
  SECOND_SCREEN,
  THIRD_SCREEN,
  FOURTH_SCREEN,
  FIFTH_SCREEN,
} from './constants';

import messages from './messages';

/* eslint jsx-a11y/no-noninteractive-element-interactions: 0 */

const Header = ({ showLoginModal, account }) => {
  const [isToggled, setToggled] = useState(false);

  function toggle(screen) {
    setToggled(!isToggled);
    scrollToSection(typeof screen === 'string' ? `#${screen}` : ``);
  }

  return (
    <Wrapper>
      <Box id={HEADER_ID} isToggled={isToggled}>
        <div className="container">
          <div className="d-flex flex-lg-row flex-column justify-content-between">
            <div className="d-flex justify-content-between flex-grow-1 flex-lg-grow-0">
              <A to={routes.questions()} className="logo">
                <img src={logo} alt="logo" />
              </A>

              <button
                className="d-inline-block d-lg-none navbar-toggler"
                type="button"
                onClick={toggle}
              >
                <Icon
                  icon={isToggled ? closeIcon : headerNavigation}
                  width="20"
                  noMargin
                />
              </button>
            </div>

            <div
              className={`${
                isToggled ? '' : 'd-none'
              } d-lg-flex flex-column flex-lg-row navbar`}
            >
              <button onClick={() => toggle(SECOND_SCREEN)}>
                <FormattedMessage {...messages.about} />
              </button>

              <button onClick={() => toggle(THIRD_SCREEN)}>
                <FormattedMessage {...messages.rewards} />
              </button>

              <button onClick={() => toggle(FOURTH_SCREEN)}>
                <FormattedMessage {...messages.faq} />
              </button>

              <button onClick={() => toggle(FIFTH_SCREEN)}>
                <FormattedMessage {...messages.team} />
              </button>

              {!account && (
                <React.Fragment>
                  <button className="login" onClick={showLoginModal}>
                    <Icon className="mr-2" width="20" icon={login} noMargin />
                    <FormattedMessage {...messages.login} />
                  </button>

                  <Button
                    className="signup"
                    onClick={() =>
                      createdHistory.push(routes.signup.email.name)
                    }
                  >
                    <FormattedMessage {...messages.signUpFree} />
                  </Button>
                </React.Fragment>
              )}

              {account && (
                <Button
                  className="signup"
                  onClick={() => createdHistory.push(routes.questions())}
                >
                  <FormattedMessage {...messages.goToSite} />
                </Button>
              )}
            </div>
          </div>
        </div>
      </Box>
    </Wrapper>
  );
};

const Box = styled.div`
  position: relative;
  top: 0;
  width: 100%;
  background-color: #17234af2;
  z-index: 9999;
  padding: 19px 0;

  .logo img {
    width: 240px;
    text-align: left;
    margin-top: 7px;
  }

  .navbar {
    padding-top: 12px;
    padding-bottom: 12px;

    button {
      outline: none;
      display: flex;
      align-items: center;
      letter-spacing: -0.9px;
      font-family: ${LANDING_FONT};
      font-weight: 400;
      font-size: 16px;
      padding-left: 20px;
      padding-right: 20px;
    }

    button:not(.signup) {
      color: ${TEXT_LIGHT};

      :hover {
        color: ${TEXT_PRIMARY};
        ${IconHover({ color: TEXT_PRIMARY })};
      }
    }

    .signup {
      margin-left: 25px;
      min-width: 140px;
    }
  }

  @media only screen and (max-width: 992px) {
    padding: 11px 0;
    position: fixed;
    background-color: #17234a;
    height: ${x => (x.isToggled ? `100vh` : `auto`)};

    .logo img {
      width: 180px;
    }

    .navbar {
      button {
        width: 100%;
        justify-content: center;
        padding: 30px 0 !important;
      }

      .signup {
        margin-left: 0px !important;
        padding: 25px 0 !important;
      }
    }
  }
`;

const Wrapper = styled.header`
  .scroll {
    position: fixed;
    background-color: rgba(9, 17, 40, 0.9);
    animation: header 1s;
    padding: 5px 0;

    @keyframes header {
      0% {
        transform: translate(0px, -180px);
      }
      100% {
        transform: translate(0, 0px);
      }
    }

    .logo img {
      width: 180px !important;
    }
  }
`;

Header.propTypes = {
  showLoginModal: PropTypes.func,
  account: PropTypes.string,
};

export default React.memo(Header);
