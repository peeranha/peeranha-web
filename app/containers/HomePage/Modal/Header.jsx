import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';

import messages from '../messages';

const Box = styled.header`
  h4 {
    padding-bottom: 35px;
    text-align: center;
  }

  @media only screen and (max-width: 576px) {
    h4 {
      padding-bottom: 15px;
    }
  }
`;

export default () => (
  <Box>
    <h4>
      <FormattedMessage {...messages.beTheFirstOne} />
    </h4>
  </Box>
);
