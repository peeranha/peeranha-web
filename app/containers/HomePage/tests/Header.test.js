import { Header } from '../Header';

jest.mock('createdHistory', () => ({
  push: jest.fn(),
}));

const cmp = new Header();
cmp.props = {
  sendEmailLoading: false,
  sendEmail: () => {},
  showHeaderPopupDispatch: jest.fn(),
  closeHeaderPopupDispatch: jest.fn(),
  showPopup: true,
  popupPosition: {},
};

describe('Header', () => {
  it('test', () => {
    expect(cmp.render()).toMatchSnapshot();
  });
});
