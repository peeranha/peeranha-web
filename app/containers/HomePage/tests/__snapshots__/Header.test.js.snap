// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test test 1`] = `
<Header__Wrapper>
  <Header__Box
    id="landing-header"
  >
    <ModalDialog
      customPosition={Object {}}
      show={true}
    >
      <div
        className="header-modal-dialog"
      >
        <div
          className="image-coins"
        >
          <img
            alt="bgLogin"
            src="IMAGE_MOCK"
          />
        </div>
        <div>
          <p
            className="modal-dialog-message"
          >
            <FormattedMessage
              defaultMessage="The platform is under developing, but you can be the first to join and earn up to 100x more."
              id="app.containers.HomePage.platformUnderDeveloping"
              values={Object {}}
            />
          </p>
          <ReduxForm
            button={
              Object {
                "defaultMessage": "Get Reward",
                "id": "app.containers.HomePage.getReward",
              }
            }
            form="HEADER"
            sendEmail={[Function]}
            sendEmailLoading={false}
          />
        </div>
      </div>
    </ModalDialog>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-6 col-xl-6 col-lg-4 logo"
        >
          <button
            data-hash="fs1"
            onClick={[Function]}
          >
            <img
              alt="logo"
              src="IMAGE_MOCK"
            />
          </button>
        </div>
        <button
          className="col-6 d-inline-block d-lg-none navbar-toggler navbar-dark"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="col-md-12 col-xl-6 col-lg-8 nav-bar d-none d-lg-block"
          id="navbartogglerId"
        >
          <div
            className="row"
          >
            <div
              className="col-md-12 col-lg-7"
            >
              <div
                className="row navigation"
              >
                <button
                  className="col-md-12 col-lg-3"
                  data-hash="fs2"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    defaultMessage="About"
                    id="app.containers.HomePage.about"
                    values={Object {}}
                  />
                </button>
                <button
                  className="col-md-12 col-lg-3"
                  data-hash="fs3"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    defaultMessage="Rewards"
                    id="app.containers.HomePage.rewards"
                    values={Object {}}
                  />
                </button>
                <button
                  className="col-md-12 col-lg-3"
                  data-hash="fs4"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    defaultMessage="FAQ"
                    id="app.containers.HomePage.faq"
                    values={Object {}}
                  />
                </button>
                <button
                  className="col-md-12 col-lg-3"
                  data-hash="fs5"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    defaultMessage="Team"
                    id="app.containers.HomePage.team"
                    values={Object {}}
                  />
                </button>
              </div>
            </div>
            <div
              className="col-md-12 col-lg-5"
            >
              <div
                className="row"
              >
                <button
                  className="col-md-12 col-lg-6 log-in-button"
                  onClick={[Function]}
                >
                  <img
                    alt="login"
                    src="IMAGE_MOCK"
                  />
                  <FormattedMessage
                    defaultMessage="Log in"
                    id="app.containers.HomePage.login"
                    values={Object {}}
                  />
                </button>
                <button
                  className="col-md-12 col-lg-6 sign-up-button"
                  onClick={[Function]}
                >
                  <FormattedMessage
                    defaultMessage="Sign up free"
                    id="app.containers.HomePage.signUpFree"
                    values={Object {}}
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </Header__Box>
</Header__Wrapper>
`;
